Automattic/jetpack
# Eslint: Ignore existing warnings when run via `yarn lint`

#### Changes proposed in this Pull Request:

I couldn't find a tool like phpcs-changed for eslint at a quick search,
so I worked out something that should serve our purposes: a "formatter"
for eslint that ignores the warnings that we want to ignore.

It's intentionally a bit fragile: if you add or delete a line in one of
these files, it'll change the line numbers for all the following
warnings and so prompt you to fix them all.

It also intentionally doesn't apply this filtering to `yarn lint-file`.
If you're linting an individual file, hopefully you'll want to fix the
warnings in it.

This also adds some paths to .eslintignore, and sets `yarn lint` to
return a failure status for warnings now that it's not reporting any
warnings on master.

#### Jetpack product discussion


p1607031642022900-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Run `yarn lint`. It shouldn't produce any warnings.
* Also the many "Unchanged files with check annotations" Github has been reporting should be gone.

#### Proposed changelog entry for your changes:


* None needed.
